Redesigned container storage setup

rhvgoyal opened the pull request:
These are patches for new container-storage-setup design.

This gets rid of recently introduced mode of user specified INPUTFILE. Newly introduced commands should be used instead.

Old mode (comaptibility mode) will continue to work as it is. This mode assumes inputfile to be /etc/sysconfig/docker-storage-setup and output file as /etc/sysconfig/docker-storage.

rhvgoyal commented:
@rhatdan PTAL.

rhvgoyal commented:
@shishir-a412ed PTAL

rhvgoyal commented:
@rhatdan, don't merge it yet. I am working on some fixes. Pushed patches early to get some feedback.

rhvgoyal commented:
Fixed couple of issues and pushed patches again.

rhatdan commented on container-storage-setup.1:
```diff
@@ -6,15 +6,19 @@ container\-storage\-setup - Tool for setting up storage for container runtimes.
 .SH DESCRIPTION
-  container-storage-setup reads INPUTFILE to create storage and generate
-  OUTPUTFILE for use by container runtimes.
+  container-storage-setup sets up storage for use by container
```
s/sets up/configures

rhatdan commented on container-storage-setup.1:
```diff
@@ -6,15 +6,19 @@ container\-storage\-setup - Tool for setting up storage for container runtimes.
-  Note: executing container-storage-setup without options will default to
-  using docker config files /etc/sysconfig/docker-storage-setup for
-  input and /etc/sysconfig/docker-storage for output.
+  First form of the command defaults to using docker config files
```
container-storage-setup without specifying a command defaults to using ...

rhatdan commented on container-storage-setup.1:
```diff
@@ -6,15 +6,19 @@ container\-storage\-setup - Tool for setting up storage for container runtimes.
+  /etc/sysconfig/docker-storage-setup for input and
+  /etc/sysconfig/docker-storage for output.
+
+  In the second form, it creates and manages storage configurations.
```
container-storage-setup with a command creates and manages ...

rhatdan commented on container-storage-setup.1:
```diff
@@ -6,15 +6,19 @@ container\-storage\-setup - Tool for setting up storage for container runtimes.
+  /etc/sysconfig/docker-storage for output.
+
+  In the second form, it creates and manages storage configurations.
+  Currently supported commands are create, remove, list, activate,
```
Commands listed below.  Don't maintain the list here, otherwise we will end up with bugs.

rhatdan commented on container-storage-setup.1:
```diff
@@ -33,6 +37,26 @@ command is not always sufficient to cleanup your
+
+\f[B]add-dev\f[] Add block device to storage configuration
+
+\f[B]export\f[] Export output file of storage configuration
```
Export file which can be used to set environment variables for use by container runtimes?

rhvgoyal replied:
ok

rhatdan commented on container-storage-setup.1:
```diff
@@ -33,6 +37,26 @@ command is not always sufficient to cleanup your
+
+\f[B]remove\f[]	Remove storage configuration
+
+\f[B]list\f[] List storage configuration
```
List currently created storage configurations.

rhatdan commented on container-storage-setup.1:
```diff
@@ -41,6 +65,17 @@ their default settings. Anything user wants to change, should be changed
+.B container-storage-setup create -o OUTPUTFILE example-config INPUTFILE
+
+Above will create a storage configuratoin named examle-config as
+specified in INPUTFILE and put the output in OUTPUTFILE.
```
Need a better description/name of OUTPUTFILE.

rhatdan commented:
Need updated README.md file but we can do this in a different pull request.

rhvgoyal commented:
@rhatdan Took care of your comments on man page. PTAL.

Also rebased on top of latest. Renamed my tests as shishir had introduced tests with similar numbers. Also renamed some of the functions.

TomSweeneyRedHat commented on container-storage-setup.1:
```diff
@@ -41,6 +64,22 @@ their default settings. Anything user wants to change, should be changed
 in the INPUTFILE. This is the file which will
 override any settings specified in /usr/share/container-storage-setup/container-storage-setup.
 
```
Suggested nit fix.  Change 'configuration in INPUTFILE or /etc/...' to 'configuration in the INPUTFILE or in /etc/...'.  Also "Anything user wants to change,' to 'Anything that the user wants to change'

rhvgoyal replied:
will do.

TomSweeneyRedHat commented on container-storage-setup.1:
```diff
@@ -41,6 +64,22 @@ their default settings. Anything user wants to change, should be changed
+
+.B container-storage-setup create -o OUTPUTFILE example-config INPUTFILE
+
+Above will create a storage configuratoin named examle-config as
```
example-config, not examle-config

rhvgoyal replied:
good catch. will fix.

TomSweeneyRedHat commented on container-storage-setup.1:
```diff
@@ -41,6 +64,22 @@ their default settings. Anything user wants to change, should be changed
+.B container-storage-setup create -o OUTPUTFILE example-config INPUTFILE
+
+Above will create a storage configuratoin named examle-config as
+specified in INPUTFILE and put the output in OUTPUTFILE. OUTPUTFILE is
```
'and will put the output'

rhvgoyal replied:
ok

TomSweeneyRedHat commented on container-storage-setup.1:
```diff
@@ -41,6 +64,22 @@ their default settings. Anything user wants to change, should be changed
+Above will create a storage configuratoin named examle-config as
+specified in INPUTFILE and put the output in OUTPUTFILE. OUTPUTFILE is
+a file which can be parsed by container runtime for various config
+options. In a typical form, output file can be passed in
```
'passed in the'

rhvgoyal replied:
ok

TomSweeneyRedHat commented on container-storage-setup.sh:
```diff
@@ -694,6 +747,47 @@ canonicalize_block_devs() {
+# This is used in config creation mode
+canonicalize_block_devs_generic() {
+  local devs=$1 dev
+  local devs_resolved dev_resolved
```
I don't have a great suggestion, but would ask you to consider renaming devs_resolved or dev_resolved.  They are so similar in naming that I could see them becoming confused or mistyped in later maintenance.  Perhaps simply devs_resolved to resolved_devs just for the sake of differentiation?

rhatdan replied:
local devs_resolved resolved_device?

TomSweeneyRedHat replied:
👍

rhvgoyal replied:
ok, will change.

rhvgoyal commented:
@TomSweeneyRedHat I have taken care of your comments and pushed new patches. PTAL.

rhvgoyal commented:
@rhatdan PTAL

rhatdan commented:
@rh-atomic-bot r+ f195fc7
